pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -86,9 +86,39 @@ func (vim *virtIOInterfaceManager) hotplugVirtioInterface(vmi *v1.VirtualMachine
 	return nil
 }
 
-func domainInterfaceFromNetwork(domain *api.Domain, network v1.Network) *api.Interface {
-	for _, iface := range domain.Spec.Devices.Interfaces {
-		if iface.Alias.GetName() == network.Name {
+func (vim *virtIOInterfaceManager) hotUnplugVirtioInterface(vmi *v1.VirtualMachineInstance, currentDomain *api.Domain, updatedDomain *api.Domain) error {
+	for _, domainIface := range interfacesToHotUnplug(vmi.Spec.Domain.Devices.Interfaces, currentDomain.Spec.Devices.Interfaces) {
+		log.Log.Infof("preparing to hot-unplug %s", domainIface.Alias.GetName())
+
+		ifaceXML, err := xml.Marshal(domainIface)
+		if err != nil {
+			return err
+		}
+
+		if derr := vim.dom.DetachDeviceFlags(strings.ToLower(string(ifaceXML)), affectLiveAndConfigLibvirtFlags); derr != nil {
+			log.Log.Reason(derr).Errorf("libvirt failed to detach interface %s: %v", domainIface.Alias.GetName(), derr)
+			return derr
+		}
+	}
+	return nil
+}
+
+func interfacesToHotUnplug(vmiSpecInterfaces []v1.Interface, domainSpecInterfaces []api.Interface) []api.Interface {
```

Comment:
Interfaces with ordinal naming scheme should be ignored (the domain interface target is the tap device name, it should contain this info).

Reply:
Done